libshaderc_spvc/src/spvc_private.cc

```diff
@@ -288,14 +289,50 @@ shaderc_spvc_compilation_result_t generate_hlsl_shader(
     const uint32_t* source, size_t source_len,
     shaderc_spvc_compile_options_t options,
     shaderc_spvc_compilation_result_t result) {
-  std::unique_ptr<spirv_cross::CompilerHLSL> compiler(
-      new (std::nothrow) spirv_cross::CompilerHLSL(source, source_len));
+
+  std::unique_ptr<spirv_cross::CompilerHLSL> compiler;
+
+// spvc IR generation is under development, for now run spirv-cross
+// compiler(SHADERC_ENABLE_SPVC_PARSER is OFF by default)
+// TODO (sarahM0): change the default to spvc IR generation when it's done
+#if SHADERC_ENABLE_SPVC_PARSER
+  spvtools::Optimizer opt(options->source_env);
+  opt.SetMessageConsumer(std::bind(
+      consume_spirv_tools_message, result, std::placeholders::_1,
+      std::placeholders::_2, std::placeholders::_3, std::placeholders::_4));
+
+  spirv_cross::ParsedIR ir;
+  ir.spirv = std::vector<uint32_t>(source, source + source_len);
+
+  opt.RegisterPass(
+      spvtools::Optimizer::PassToken(std::unique_ptr<spvtools::opt::Pass>(
+          reinterpret_cast<spvtools::opt::Pass*>(
+              new spvtools::opt::SpvcIrPass(ir)))));
+
+  if (!opt.Run(source, source_len, &result->binary_output)) {
+    result->messages.append(
+        "Transformations between source and target "
+        "execution environments failed.\n");
+    result->status = shaderc_compilation_status_transformation_error;
+    compiler.reset(new (std::nothrow) spirv_cross::CompilerHLSL(ir));
+  }
+
```

Comment:
If run is successful, will there me more after the spvc-ir is generated?

Reply:
Yes you're right. The plan is to replace the failure message with this:
compiler.reset(new spirv_cross::CompilerHLSL(ir));